### PR TITLE
Payment Type (Ticket #3)

### PR DESCRIPTION
**Pull request by joeywellman**
# Description

Added IsArchived Boolean to Payment Type Model (PaymentTypeController.cs).

Created Controller for Payment Type Resource (PaymentTypeController.cs) and Testing Class / Methods (TestPaymentType.cs) that Validate Operations:
1. GET All
2. GET Single
3. POST
4. PUT
5. DELETE (Soft/Hard)

## Related Ticket(s)
Ticket 3:
https://trello.com/c/4l2BKwUk/3-ticket-3-allow-developers-to-access-payment-type-resource

## Steps to Test

1. git add and git commit whatever you're working on (on your branch)
2. git checkout master, to switch to our local master branch
3. git fetch --all will give you access to my branch
4. git checkout jw-paymenttype to switch to my branch
5. If you don't see the most recent version of my files, try git pull origin jw-paymenttype
6. start BangazonAPI.sln
7. CTRL+R, A to run tests
8. Use SSMS / SQL Viewer to SQL query API ("SELECT * FROM PaymentType")
9. If something is crab-hammered, tell me - don't write any code on my branch
10. Comment on my PR to tell me "great job" or "you fail" and why

# Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: BangazonAPI/BangazonAPI/Controllers/PaymentTypeController.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        public async Task<IActionResult> PostPaymentType([FromBody] PaymentType paymentType)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = $@"INSERT INTO PaymentType (AcctNumber, Name, CustomerId, IsArchived)
+                                                    OUTPUT INSERTED.Id
+                                                    VALUES (@AcctNumber, @Name, @CustomerId, 1)";
```
Comment (jordan-castelloe) [inline BangazonAPI/BangazonAPI/Controllers/PaymentTypeController.cs]: So this is exactly the right idea, except that the bit of 1 converts to `true` (`false` is 0), which means that whenever we're creating a new payment type it automatically sets `isArchived` to `true`. We want it to start out as false and change to true when they delete it.

**Review by jordan-castelloe — DISMISSED** [review]
One tiny comment, otherwise looks great!

**Review by jordan-castelloe — CHANGES_REQUESTED** [review]
There's a problem with the `POST` method but otherwise everything looks great!

**Comment by Havey3** [comment]
Testing all worked good
Was able to get all payment types and single payment type
Looks good once the isarchived is set to false 👍

**Comment by MattRowee** [comment]
🔪 🌇 🔪 , knife city

File: BangazonAPI/BangazonAPI/Controllers/PaymentTypeController.cs (hunk 1 above)
Comment (joeywellman) [reply BangazonAPI/BangazonAPI/Controllers/PaymentTypeController.cs]: Thanks!  Made applicable changes and updated to [current version of] master!

**Review by jordan-castelloe — APPROVED** [review]

**Pull request merged**